src/navigator-card-body.jsx

```diff
@@ -410,34 +480,33 @@
         />
     );
 
-    if (isGrid) {
-        return (
-            <>
-                {contextMenu}
-                <CardBody
-                  id="navigator-card-body"
-                  onClick={resetSelected}
-                  onContextMenu={handleContextMenu}
-                >
-                    <Gallery id="folder-view">
-                        {sortedFiles.map(file => <Item file={file} key={file.name} />)}
-                    </Gallery>
-                </CardBody>
-            </>
-        );
-    } else {
-        return (
-            <>
-                {contextMenu}
-                <ListingTable
-                  id="folder-view"
-                  className="pf-m-no-border-rows"
-                  variant="compact"
-                  columns={[_("Name")]}
-                  rows={sortedFiles.map(file => ({ columns: [{ title: <Item file={file} key={file.name} /> }] }))}
-                  onContextMenu={handleContextMenu}
-                />
-            </>
-        );
-    }
+    return (
+        <>
+            {contextMenu}
+            <div id={navigator_parent_id}>
+                {isGrid &&
+                    <CardBody
+                      id="navigator-card-body"
+                      onClick={handleClick}
+                      onDoubleClick={handleDoubleClick}
+                      onContextMenu={handleContextMenu}
+                    >
+                        <Gallery id="folder-view">
+                            {sortedFiles.map(file => <Item file={file} key={file.name} />)}
+                        </Gallery>
+                    </CardBody>}
+                {!isGrid &&
+                    <ListingTable
+                      id="folder-view"
+                      className="pf-m-no-border-rows"
+                      variant="compact"
+                      columns={[_("Name")]}
+                      rows={sortedFiles.map(file => ({ columns: [{ title: <Item file={file} key={file.name} /> }] }))}
+                      onDoubleClick={handleDoubleClick}
+                      onContextMenu={handleContextMenu}
+                      onClick={handleClick}
```

Comment:
These 9 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-249-20240207-125527-92469164-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)